dolphin/drivers/manager.py

```diff
@@ -11,18 +11,31 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import threading
 
 from oslo_log import log
 
 
 LOG = log.getLogger(__name__)
+lock = threading.Lock()
```

Comment:
Why this lock is global and cannot be the class variable? Please put a comment

Reply:
I put it as a global variable to ensure that every instances share the same lock.
If I put it as a class variable, there would be a problem: every class that inherited from class Singleton would have the same lock, for example, if TaskManager and DriverManager all inherited from Singleton, TaskManager and DriverManager would share the same lock, that is not what we want.